_posts/2015-08-31-intro-to-daisywheeljs.markdown

```diff
@@ -0,0 +1,43 @@
+---
+layout: post
+title: Adding the DaisywheelJS onscreen-keyboard to HTML5 Gamepad projects
+tags: daisywheeljs daisywheel gamepad html5
+username: likethemammal
+---
+
+### Some background
+
+HTML5 gaming has developed quite a bit since it first landed on the scene a few years back. Now, with proclamations from major browser vendors starting to pull support from Flash, and even [block some Flash ads(http://www.themonitordaily.com/google-chrome-adobe-autoplay/25687/), usage of HTML5’s latest technologies will spread further. Old Flash game devs will become new HTML5 game devs and entire games will be ported over for maintainability’s sake.
+
+The [Gamepad API](https://w3c.github.io/gamepad/) is one of those technologies. It's been mostly ignored by the web industry, but it’s extremely exciting to those that want the browser to become the next console-like gaming experience.
+
+One of the things console developers still haven’t gotten right though, is the on-screen keyboard. You know, that awful thing they shove in your face whenever you need to enter your login, or your name for a high-score. This is the same QWERTY layout that’s present on your desktop keyboard, and was originally invented for typewriters (no joke, [it’s true](https://en.wikipedia.org/wiki/QWERTY#History_and_purposes)). For a user with a controller in their hand this is beyond frustrating, and drags them from that precious immersion game designers strive for so religiously.
+
+### A solution appears!
+
+If you’re a user of Steam, the massive game marketplace that has taken over PC gaming, you might have used their [Big Picture Mode](http://store.steampowered.com/bigpicture/). It’s an amazing tool and gives me more hope for the future. Basically, it turns the Steam interface into something that resembles a console and is much more controller friendly. This compared to the normal layout navigated with a mouse and keyboard. The “friends” tab of Big Picture Mode presents you with an on-screen keyboard whenever you chat. This has been deemed the “Daisywheel” (after the [old impact printing method](https://en.wikipedia.org/wiki/Daisy_wheel_printing)) And this is where our journey begins.
+
+Imagine a radial keyboard, with groups of symbols around the edge, requiring only 2 “actions” (from the user) to select any character or number. This, opposed to the possible 12 “actions” that a traditional QWERTY on-screen keyboard might require (furiously arrowing from letter to letter). Just to select a single character.
```

Comment:
Traditional keyboards implemented for typing on controllers present the user with a big, full-sized QUERTY keyboard. In order to type out a message, the user has to navigate around the keyboard with the thumb-sticks on their controller, vigorously inputting anywhere from 2 to 12 actions just to enter a single character. Now imagine a world where a radial keyboard existed, with groups of symbols around the edge, requiring only 2 actions from the user (a flick of the thumb-stick and a tap on an action button) to select any character or number.